src/Streamly/Prelude.hs

```diff
@@ -303,6 +315,10 @@ head m =
         yield a _ = return (Just a)
     in (runStream (toStream m)) Nothing stop yield
 
+-- | Returns the stream minus the head.
+tail :: Streaming t => t m a -> t m a
```

Comment:
The `tail` might be unnecessary. I was thinking of the standard Prelude. Just checked Pipes Prelude and Conduits, nothing like this exists.